src/metabase/models/card.clj

```diff
@@ -873,6 +874,62 @@ saved later when it is ready."
                (map? (:dataset_query card)) (update :dataset_query dissoc :lib/metadata))]
     (next-method card json-generator)))
 
+(defn- replaced-inner-query-for-native-card
+  [query {:keys [fields tables] :as _replacement-ids}]
+  (let [id->field           (if (empty? fields)
+                              {}
+                              (group-by :id
+                                        (t2/query {:select [[:f.id :id]
+                                                            [:f.name :column]
+                                                            [:t.name :table]
+                                                            [:t.schema :schema]]
+                                                   :from   [[:metabase_field :f]]
+                                                   :join   [[:metabase_table :t] [:= :f.table_id :t.id]]
+                                                   :where  [:in :f.id (set/union (into #{} (keys fields))
+                                                                                 (into #{} (vals fields)))]})))
```

Comment:
I feel like I'm always terrible at writing Toucan queries. Surely there's a better way to write…this entire function?

Reply:
Well... I don't know if Toucan can do joins at all, but for a normal honeysql query this looks ok? Except I'd use `m/index-by` rather than `group-by`, but the query itself is fine?